feat: Add new domain/device type dropdown for Web Analytics + filters UI revamp ✨

rafaeelaudibert opened the pull request:
## Problem

We want a domain dropdown on web analytics! It'll look much nicer.

## Changes

There are a lot of changes. I'm not hiding anything behind a FF because the change is pretty big, and I didn't want to keep the old code around. I had to change a lot of it.

Desktop view is nicer, tablet view is much better, mobile view is 2000% better. It feels like an actual thought-out navbar now - I put way too much thought into it, 2 days to build this.

See video to understand what it looks like now, snapshots should update too

https://www.loom.com/share/e454b9890a8f4397b685e326ef1575bc?sid=31ca6ef8-d45b-4a63-b3b4-9df3d5d48c9b

![image](https://github.com/user-attachments/assets/2eadb4c1-910d-456f-9ac7-980eb30c56da)


## Does this work well for both Cloud and self-hosted?

Yep

## How did you test this code?

Manually, plus snapshots will update

rafaeelaudibert committed:
refactor: Extract components to separate files

Just organize this slightly better

rafaeelaudibert committed:
feat: Add new settings for authorized URLs

We're now making the authorized URLs a slightly higher level configuration because this will be more important now that you'll be able to filter by it in Web Analytics

rafaeelaudibert committed:
feat: Add warning to Web Analytics if URLs aren't set

rafaeelaudibert committed:
refactor: Add some more unwanted `INACTIVE_FLAGS`

These change the UI too heavily and I don't want them to happen locally, so let's turn them off locally by default by adding them to `INACTIVE_FLAGS`

rafaeelaudibert committed:
fix: Render bigger path cleaning toggle

rafaeelaudibert committed:
refactor: Huge refactor to Web Analytics navbar

There's still a little bit missing, but it's almost there, see TODOs

rafaeelaudibert committed:
feat: Hide Web Analytics filters behind popover

This will clean up some space for web analytics on middle-sized screens. We're also further hiding it behind a button on even smaller screens (mobile only)

rafaeelaudibert committed:
feat: Make domain dropdown actually work

rafaeelaudibert committed:
fix: Avoid popover going out of bounds on mobile

robbie-c commented:
<img width="316" alt="Screenshot 2025-02-24 at 11 14 53" src="https://github.com/user-attachments/assets/78d820f2-d366-4cff-81d7-1d3560ae9bff" />

It'd be cool to have this link to where I can set up hosts. This is probably the first place people will encounter this feature, so it'd be good to give them an action to take if they want more sites on this list!

robbie-c commented:
Slightly tricky edge case here, I tried to add a host via the filters UI

<img width="918" alt="Screenshot 2025-02-24 at 11 17 54" src="https://github.com/user-attachments/assets/b33c568f-57d8-4949-9c6d-273065be0261" />

nothing seemed to happen. I think this is due to how you are filtering out host filters from this list.

robbie-c commented:
<img width="1258" alt="Screenshot 2025-02-24 at 11 21 43" src="https://github.com/user-attachments/assets/2bc0c531-0035-45a5-a018-88eb42b012ea" />

Ah this doesn't work on the demo env because we don't initialize propdefs in the demo env setup.

I think this is fine, but if you were super keen you could make it do this. Orrrrr, you could simultaneously query events from last few hours and grab some candidate hostnames that way

rafaeelaudibert commented:
@robbie-c 

> It'd be cool to have this link to where I can set up hosts. This is probably the first place people will encounter this feature, so it'd be good to give them an action to take if they want more sites on this list!

Didn't you have a warning about you having no set-up domains? I set one up like the ones we have for missing `$pageview` events. I'll try and improve on that, though, agreed they'll need to set this up. I'm gonna be adding setting that up to the onboarding flow as well (follow-up PR) so that should also help

> Slightly tricky edge case here, I tried to add a host via the filters UI

Yeah, aware, I'll add a way to "disable" specific filters from being added, that doesn't exist yet, so I chose to just let it die silently

> I think this is fine, but if you were super keen you could make it do this. Orrrrr, you could simultaneously query events from last few hours and grab some candidate hostnames that way

That's what I'm doing actually, I wonder why that doesn't work for you? I'm seeing `hedgebox` as a suggestion for example

> I found this UI a bit confusing, I started type in the search box to add a domain. I don't think it makes sense to have the search bar and the add button on the same row given that they aren't related.

Oh yeah, good call. I just reused the UI we have, but it needs to be improved before we launch this, let me take that UI for a ride

rafaeelaudibert commented on frontend/src/scenes/web-analytics/WebPropertyFilters.tsx:
```diff
@@ -11,17 +16,54 @@ export const WebPropertyFilters = ({
+                            TaxonomicFilterGroupType.SessionProperties,
+                        ]}
+                        onChange={(filters) =>
+                            // We want to ignore `$host` filters, they're controlled by the domain filter
+                            // If this gets confusing, we'll need to find a way to block these filters from being added
+                            setWebAnalyticsFilters(
+                                filters
+                                    .filter(isEventPersonOrSessionPropertyFilter)
+                                    .filter((event) => event.key !== '$host')
+                            )
+                        }
+                        propertyFilters={propertyFilters}
```
I tried, Typescript complained

rafaeelaudibert committed:
refactor: Small Greptile-suggestions cleanup

rafaeelaudibert committed:
feat: Link to where new domains can be added

rafaeelaudibert committed:
refactor: Better titles

rafaeelaudibert committed:
refactor: Remove authorized URL search

It's not immediately useful and it's making the UI slightly confusing, so let's remove that

rafaeelaudibert committed:
feat: Hide `host` as an option inside Web Analytics filter

I'm not sure this is the right approach, we might want this one to be added in here just fine but I'll discuss this on the PR

rafaeelaudibert commented:
@robbie-c @lricoy ready for the second round. Pay attention to dd598ac6c6590904059a0deac565fd60d3fa9c65. Do you expect `$host` to not be filterable because we already have a higher-level `$host` filter or is that too restrictive and we should let people compound them?

rafaeelaudibert committed:
refactor: Display `HeatmapButton` only when domain is selected

rafaeelaudibert committed:
refactor: Extract `WebAnalyticsFilter` to separate file

rafaeelaudibert committed:
refactor: Extract all filters to separate file

Simplify display logic by letting each individual filter control when they are rendered

rafaeelaudibert committed:
feat: Add mobile/desktop selector to WA/WV

rafaeelaudibert committed:
fix: Fix TS error

lricoy commented:
> @robbie-c @lricoy ready for the second round.
> 
> ~Pay attention to [dd598ac](https://github.com/PostHog/posthog/commit/dd598ac6c6590904059a0deac565fd60d3fa9c65). Do you expect `$host` to not be filterable because we already have a higher-level `$host` filter or is that too restrictive and we should let people compound them? ~ I've decided to remove that restriction and allow people to filter by `$host`

I think hiding the host filter option there is a sensible choice. Maybe adding a informative note on the "empty result" or disabling it instead could also work but I would go with hiding for now to avoid impossible states.

rafaeelaudibert committed:
fix: Fix skeleton height

rafaeelaudibert committed:
fix: Fix colors broken on canvas

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

rafaeelaudibert committed:
fix: Properly left-align web vitals loading bar

And make it occupy full width on non-web-vitals

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

robbie-c commented:
> Oh yeah, good call. I just reused the UI we have, but it needs to be improved before we launch this, let me take that UI for a ride

Ha nice one!

> I've decided to remove that restriction

I still found some slightly weird interactions here. I managed to get into a state where I have a host filter set, but it doesn't show up in either the domain chooser UI or the filters UI. The only way I know it's set is that it's still in the URL. Here's a link: http://localhost:8000/project/1/web?filters=%5B%7B%22key%22%3A%22%24host%22%2C%22value%22%3A%5B%22hedgebox.net%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%2C%7B%22type%22%3A%22event%22%2C%22key%22%3A%22%24host%22%2C%22value%22%3A%22all%22%2C%22operator%22%3A%22exact%22%7D%5D&path_cleaning=false&filter_test_accounts=false&compare_filter=%7B%22compare%22%3Atrue%7D&domain=all

I'm happy to merge though, let's keep this moving forward especially with a diff this big
